commit 78f3a4417f49248ab4a9ef18fec7254b99440ca6
Author: Haozhong Zhang <haozhong.zhang@intel.com>
Date:   Tue May 3 18:50:07 2016 +0800

    lpc_ich9: set LPC pm_base to 0x600 for pc-lite
    
    Traditionally, pm_base is set by guest BIOS. However, pc-lite does not
    use any guest BIOS. Therefore, we have to setup it in qemu. The pm_base
    0x600 is the value used in SeaBIOS and qboot.
    
    Signed-off-by: Haozhong Zhang <haozhong.zhang@intel.com>

diff --git a/hw/isa/lpc_ich9.c b/hw/isa/lpc_ich9.c
index 99cd3ba..5a3e925 100644
--- a/hw/isa/lpc_ich9.c
+++ b/hw/isa/lpc_ich9.c
@@ -480,6 +480,9 @@ static void ich9_lpc_reset(DeviceState *qdev)
     ICH9LPCState *lpc = ICH9_LPC_DEVICE(d);
     uint32_t rcba_old = pci_get_long(d->config + ICH9_LPC_RCBA);
     int i;
+    PCMachineState *pcms = PC_MACHINE(qdev_get_machine());
+    PCMachineClass *pcmc = PC_MACHINE_GET_CLASS(pcms);
+    uint32_t pm_base = ICH9_LPC_PMBASE_DEFAULT;
 
     for (i = 0; i < 4; i++) {
         pci_set_byte(d->config + ICH9_LPC_PIRQA_ROUT + i,
@@ -491,7 +494,10 @@ static void ich9_lpc_reset(DeviceState *qdev)
     }
     pci_set_byte(d->config + ICH9_LPC_ACPI_CTRL, ICH9_LPC_ACPI_CTRL_DEFAULT);
 
-    pci_set_long(d->config + ICH9_LPC_PMBASE, ICH9_LPC_PMBASE_DEFAULT);
+    if (pcmc->no_bios) {
+        pm_base = 0x600 | ICH9_LPC_PMBASE_RTE;
+    }
+    pci_set_long(d->config + ICH9_LPC_PMBASE, pm_base);
     pci_set_long(d->config + ICH9_LPC_RCBA, ICH9_LPC_RCBA_DEFAULT);
 
     ich9_cc_reset(lpc);
